tests/python_package_test/test_engine.py

```diff
@@ -464,21 +465,21 @@ def test_cv(self):
                                           '../../examples/lambdarank/rank.train.query'))
         params_lambdarank = {'objective': 'lambdarank', 'verbose': -1, 'eval_at': 3}
         lgb_train = lgb.Dataset(X_train, y_train, group=q_train)
-        # ... with NDCG (default) metric
-        cv_res_lambda = lgb.cv(params_lambdarank, lgb_train, num_boost_round=10, nfold=3,
-                               verbose_eval=False)
-        self.assertEqual(len(cv_res_lambda), 2)
-        self.assertFalse(np.isnan(cv_res_lambda['ndcg@3-mean']).any())
         # ... with l2 metric
         cv_res_lambda = lgb.cv(params_lambdarank, lgb_train, num_boost_round=10, nfold=3,
                                metrics='l2', verbose_eval=False)
         self.assertEqual(len(cv_res_lambda), 2)
         self.assertFalse(np.isnan(cv_res_lambda['l2-mean']).any())
+        # ... with NDCG (default) metric
```

Comment:
Switching `# ... with l2 metric` and `# ... with NDCG (default) metric` is needed to ensure that `params_lambdarank` is not modified during cv.